blocks/http/include/gnuradio-4.0/http/HttpBlock.hpp

```diff
@@ -220,6 +223,8 @@ class HttpBlock : public gr::Block<HttpBlock<T>, BlockingIO<false>, HttpBlockDoc
     gr::http::RequestType type = gr::http::RequestType::GET;
     std::string           parameters; // x-www-form-urlencoded encoded POST parameters
 
+    explicit HttpBlock() {}
```

Comment:
This slipped through in an earlier review. Blocks should not define their own constructor but rely on either the life-cycle callback methods and/or `settingsChanged(...)` for initialising. 

These constructors are not invoked by the default `emplaceBlock(...)` construction method and notably also do not invoke the constructors defined in `Block<T>` which causes issues w.r.t. settings handling, resetting blocks, and default initialisation.

If you could fix this and the corresponding unit-test please.

Reply:
There are quite a few blocks that do the same. I'd propose creating a task to remove the `emplaceBlock` version that is not settings-based (the one that is marked for removal) and to port all blocks and tests to it.